File: src/main/kotlin/inflearn/java/ch01_string/_1_3_문장_속_단어.java
package inflearn.java.ch01_string;

import java.util.Scanner;

public class _1_3_문장_속_단어 {
    public static String solution(String str) {
        String[] strArray = str.split(" ");
        String max = strArray[0];
        for(String elem: strArray) {
            if(elem.length() > max.length())
                max = elem;
        }
        return max;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        System.out.println(solution(input));
        return;
    }
}

File: src/main/kotlin/inflearn/1_3 문장 속 단어.kt
package inflearn

import java.util.*

object `1_3 문장 속 단어` {
    fun solution(str: String): String? {
        val strArray = str.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        var max = strArray[0]
        for (elem in strArray) {
            if (elem.length > max.length) max = elem
        }
        return max
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val input = sc.nextLine()
        println(solution(input))
        return
    }
}